src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -44,16 +44,30 @@ Construct a semidiscretization of a hyperbolic PDE.
 """
 function SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
                                       source_terms=nothing,
-                                      boundary_conditions=boundary_condition_periodic, RealT=real(solver))
+                                      boundary_conditions=boundary_condition_periodic,
+                                      RealT=real(solver), uEltype=RealT)
```

Comment:
I think we should also update the docstring and add an explanation for `RealT` and `uEltype` and how they differ in their purpose.

Reply:
Hm, maybe you're right. Although at some point we have so many experimental features that it might make sense to just bite the bullet and keep updating the minor release version if we decide to change stuff again.

Regarding this particular feature, we could go either way. I think documenting `RealT` is certainly useful since it is something we want to be able to change anyways. I also see the point of having an `uEltype`, though I am not yet 100% convinced of the name (although the similarity to what the DiffEq ecosphere uses it certainly a benefit). Maybe for now, only document `RealT` and leave the other maybe as a regular comment (such that our students know what to make of it)?

Reply:
Okay, sounds good. Yeah, the reason for me to pick `uEltype` was consistency with the SciML ecosystem.

Reply:
:+1:

Reply:
Done in 015b67abccd2d9fa61ded6443922af0911557f8e.